autoencoda/preprocess.py

```diff
@@ -0,0 +1,81 @@
+import argparse
+import logging
+import os
+import pickle
+
+import numpy as np
+
+parser = argparse.ArgumentParser(
+    description='Preprocess the data for my model.'
+)
+parser.add_argument('--tracks_billboard',
+                    type=str,
+                    required=False,
+                    default='../data/cache_tracks_billboard/',
+                    help='Path containing tracks that appeared on Billboard.')
+parser.add_argument('--tracks_not_billboard',
+                    type=str,
+                    required=False,
+                    default='../data/cache_tracks_not_billboard/',
+                    help='Path containing tracks that did not appear on Billboard.')
+parser.add_argument('--preprocessed',
+                    type=str,
+                    required=False,
+                    default='../data/preprocessed/',
+                    help='Path containing preprocessed data.')
+args = parser.parse_args()
+
+
+def read_pickle(full_path):
+    with open(full_path, 'rb') as f:
+        return pickle.load(f)
+
+
+def get_tracks_in_directory(path_dir):
+    tracks = []
+    dir_contents = os.listdir(path_dir)
+    # Make sure directory contains only pickled tracks.
+    assert [elem[-2:] == '.p' for elem in dir_contents]
```

Comment:
This assert just checks if the list is not None (or, in this case specifically, that it is not empty). It needs to be `assert all([elem[-2:] == '.p' for elem in dir_contents])`.

See the difference between:
`assert []`
`assert [False, True]`
`assert all([False, True])`